src/mlpack/core/dists/discrete_distribution.hpp

```diff
@@ -56,26 +56,34 @@ class DiscreteDistribution
    * @param numObservations Number of possible observations this distribution
    *    can have.
    */
-  DiscreteDistribution(const size_t numObservations) :
-      probabilities(arma::ones<arma::vec>(numObservations) / numObservations)
-  { /* nothing to do */ }
+  DiscreteDistribution(const size_t numObservations, const size_t dimension=1):
+          probabilities(dimension, numObservations)
+  {
+    probabilities.ones();
```

Comment:
Hi @rcurtin, Could I define and init a matrix like this? Why this failed and I also can't use `probabilities.set_size(dimension, numObservations)`? 

I had google it but it doesn't help. I have no idea how to use `arma:mat `class even after I read some documents about it.

Thank you